modules/core/shared/src/main/scala/lucuma/core/model/TwilightBoundedNight.scala

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model
+
+import lucuma.core.enum.Site
+import lucuma.core.math.Interval
+import lucuma.core.enum.TwilightType
+import lucuma.core.math.skycalc.TwilightCalc
+
+import cats.{ Order, Show }
+import cats.effect.Sync
+import cats.implicits._
+import monocle.Lens
+import monocle.macros.GenLens
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+/** The start and end of a particular night at a [[lucuma.core.enum.Site]]. */
+final case class TwilightBoundedNight private (
+  twilightType:     TwilightType,
+  toObservingNight: ObservingNight,
+  interval:         Interval
+) extends Night {
```

Comment:
In general, `TwilightType` and `ObservingNight` are enough to define the interval so there is opportunity to make these inconsistent.  I wonder if we should remove the redundancy?  Maybe we could move the `Interval` into a `val` calculated from the `TwilightType` and `Interval` instead of supplied as a constructor argument.

Reply:
Then there's no need to make the constructor `private` either, I think.

Reply:
This ties in with the previous comment. I completely agree but the thing is that when you take into account that given a `TwilightType`, a `Place` and a `LocalDate`, the sun may not set or rise.

This does not happen for our 2 current `Site`s. So, should we: a) Do an `unsafe` get? b) Make `interval`, `start`, `end` `Option`al and not inherit from `Night` anymore?

Reply:
Just to document here what we talked in Slack: 

- It seems that we should always allow for the creation of the object and compute the interval in-class. We could allow for empty `Interval`s. We should look into this when we evaluate switching to spire's `Interval`.
- For the moment, we are preventing inconsistencies by disallowing `copy` and not having `Lens`es, just `Getter`s.